WireAPI/Tests/WireAPITests/APIs/FeatureConfigsAPI/FeatureConfigsAPITests.swift

```diff
@@ -0,0 +1,363 @@
+        }
+    }
+
+    func testGetFeatureConfigs_FailureResponse_Insufficient_Permissions() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 403, errorLabel: "operation-denied")
+        let sut = FeatureConfigsAPIV0(httpClient: httpClient)
+
+        // Then
+        await XCTAssertThrowsError(FeatureConfigsAPIError.insufficientPermissions) {
+            // When
+            try await sut.getFeatureConfigs()
+        }
+    }
+
+    // MARK: - V4 & V5
+
+    func testGetFeatureConfigs_SuccessResponse_200_V4_TO_V5() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 200, payloadResourceName: "GetFeatureConfigsSuccessResponseV4")
+
+        let supportedVersions: [APIVersion] = [.v4, .v5]
+
+        let suts = supportedVersions.map { $0.buildAPI(client: httpClient) }
+
+        try await withThrowingTaskGroup(of: [FeatureConfig].self) { taskGroup in
+            for sut in suts {
+                taskGroup.addTask {
+                    // When
+                    try await sut.getFeatureConfigs()
+                }
+
+                for try await result in taskGroup {
+                    // Then
+                    XCTAssertEqual(
+                        result,
+                        Scaffolding.featureConfigsV4
+                    )
+                }
+            }
+        }
+    }
+
+    // MARK: - V6 and next versions
+
+    func testGetFeatureConfigs_SuccessResponse_200_V6_And_Next_Versions() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            payloadResourceName: "GetFeatureConfigsSuccessResponseV6"
+        )
+
+        let supportedVersions = APIVersion.v6.andNextVersions
+
+        let suts = supportedVersions.map { $0.buildAPI(client: httpClient) }
+
+        try await withThrowingTaskGroup(of: [FeatureConfig].self) { taskGroup in
+            for sut in suts {
+                taskGroup.addTask {
+                    // When
+                    try await sut.getFeatureConfigs()
+                }
+
+                for try await result in taskGroup {
+                    // Then
+                    XCTAssertEqual(
+                        result,
+                        Scaffolding.featureConfigsV6
+                    )
+                }
+            }
+        }
+    }
+
+}
+
+extension FeatureConfigsAPITests {
+
+    enum Scaffolding {
+
+        static func dateV4(from string: String) -> Date {
+            ISO8601DateFormatter.fractionalInternetDateTime.date(from: string)!
+        }
+
+        static func dateV6(from string: String) -> Date {
+            ISO8601DateFormatter.internetDateTime.date(from: string)!
+        }
+
+        static let featureConfigsV0: [FeatureConfig] = [
+            .appLock(.init(
+                status: .enabled,
+                isMandatory: true,
+                inactivityTimeoutInSeconds: 2_147_483_647
+            )
+            ),
```

Comment:
Perhaps indentation is a bit strange here?

Reply:
yes, I cleaned it up, thx